src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -343,7 +357,11 @@ void ResumptionDataProcessor::DeleteSubmenues(
   LOG4CXX_AUTO_TRACE(logger_);
   const uint32_t app_id = application->app_id();
   ApplicationResumptionStatus& status = resumption_status_[app_id];
-  for (auto request : status.successful_requests) {
+  auto requests = status.successful_requests;
+  requests.insert(requests.begin(),
```

Comment:
@LitvinenkoIra please add additional cycle throw the error response and internally rever it. SDL should not send additional "Delete" in case of "Add" was unsuccessful. Probably it should be not only in submenus